pkg/logql/log/metrics_extraction_test.go

```diff
@@ -152,6 +153,69 @@ func Test_Extract_ExpectedLabels(t *testing.T) {
 	require.True(t, ok)
 	require.Equal(t, (20 * time.Millisecond).Seconds(), f)
 	require.Equal(t, labels.Labels{{Name: "foo", Value: "json"}}, lbs.Labels())
+
+}
+func TestLabelExtractorWithStages(t *testing.T) {
+
+	// A helper type to check if particular logline should be skipped
+	// during `ProcessLine` or got correct sample value extracted.
+	type checkLine struct {
+		logLine string
+		skip    bool
+		sample  float64
+	}
+
+	tests := []struct {
+		name       string
+		extractor  SampleExtractor
+		checkLines []checkLine
+		shouldFail bool
+	}{
+		{
+			name: "with just logfmt and stringlabelfilter",
+			// {foo="bar"} | logfmt | subqueries != "0" (note: "0", a stringlabelfilter)
+			extractor: mustSampleExtractor(
+				LabelExtractorWithStages("subqueries", ConvertFloat, []string{"foo"}, false, false, []Stage{NewLogfmtParser(), NewStringLabelFilter(labels.MustNewMatcher(labels.MatchNotEqual, "subqueries", "0"))}, NoopStage),
```

Comment:
Why does it have the `ConvertFloat` stage? 🤔 
If the query is in string `subqueries != "0"` does it still attempts to convert to float?

Reply:
NOTE this is the `metrics_extraction` (used in metrics query not log query). 

1. ConvertFloat is to extract the value from the logline (subqueries) and convert it into sample value (other conversation methods like ConvertBytes, ConvertDuration, will treat that value differently)
2. `subqueries != "0"` is just to match the logs as a labelmatcher after `logfmt`

Both are different. Does that answer your question @ssncferreira ?